src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateAnonymousElectorDocumentIntegrationTest.kt

```diff
@@ -0,0 +1,265 @@
+        private const val OTHER_ERO_ID = "other-city-council"
+        private const val GSS_CODE = "W06000023"
+        private const val AED_SAMPLE_PHOTO = "classpath:temporary-certificate-template/sample-certificate-photo.png"
+        private const val MAX_SIZE_2_MB = 2 * 1024 * 1024
+    }
+
+    @Test
+    fun `should return forbidden given user with bearer token missing required group to access service`() {
+        wireMockService.stubCognitoJwtIssuerResponse()
+        val userGroupEroId = anotherValidEroId(ERO_ID)
+
+        webTestClient.post()
+            .uri(
+                URI_TEMPLATE,
+                ERO_ID,
+                GSS_CODE
+            )
+            .bearerToken(
+                getBearerToken(
+                    eroId = userGroupEroId,
+                    groups = listOf("ero-$userGroupEroId", "ero-vc-anonymous-admin-$userGroupEroId")
+                )
+            )
+            .contentType(MediaType.APPLICATION_JSON)
+            .withBody(buildGenerateAnonymousElectorDocumentRequest())
+            .exchange()
+            .expectStatus()
+            .isForbidden
+    }
+
+    @Test
+    fun `should return bad request given invalid request body`() {
+        // Given
+        wireMockService.stubCognitoJwtIssuerResponse()
+        val invalidGssCode = "invalid GSS Code"
+        val requestBody = buildGenerateAnonymousElectorDocumentRequest(
+            gssCode = invalidGssCode
+        )
+
+        // When
+        val response = webTestClient.post()
+            .uri(URI_TEMPLATE, ERO_ID)
+            .bearerToken(
+                getBearerToken(
+                    eroId = ERO_ID,
+                    groups = listOf("ero-$ERO_ID", "ero-vc-anonymous-admin-$ERO_ID")
+                )
+            )
+            .contentType(MediaType.APPLICATION_JSON)
+            .withBody(requestBody)
+            .exchange()
+            .expectStatus()
+            .isBadRequest
+            .returnResult(ErrorResponse::class.java)
+
+        // Then
+        val actual = response.responseBody.blockFirst()
+        assertThat(actual)
+            .hasStatus(400)
+            .hasError("Bad Request")
+            .hasMessageContaining("Validation failed for object='generateAnonymousElectorDocumentRequest'. Error count: 1")
+            .hasValidationError("Error on field 'gssCode': rejected value [$invalidGssCode], size must be between 9 and 9")
+    }
+
+    @Test
+    fun `should return not found given no local authority found for the provided gss code`() {
+        // Given
+        wireMockService.stubCognitoJwtIssuerResponse()
+        wireMockService.stubEroManagementGetEroByGssCodeNoMatch(GSS_CODE)
+        val requestBody = buildGenerateAnonymousElectorDocumentRequest(gssCode = GSS_CODE)
+
+        // When
+        val response = webTestClient.post()
+            .uri(
+                URI_TEMPLATE,
+                ERO_ID
+            )
+            .bearerToken(
```

Comment:
Not in the scope of current PR however we could introduce similar functions that we currently introduced in vca. 
`.bearerToken(getVCAdminBearerToken(eroId = ERO_ID))`

```
fun getAnonymousBearerToken(eroId: String = ERO_1234, userName: String = "an-ero-user1@$eroId.gov.uk") =
    getBearerToken(groups = listOf("ero-$eroId", "ero-vc-anonymous-admin-$eroId"), email = userName)

fun getVCAdminBearerToken(eroId: String = ERO_1234, userName: String = "an-ero-user2@$eroId.gov.uk") =
    getBearerToken(groups = listOf("ero-$eroId", "ero-vc-admin-$eroId"), email = userName)
```